### DISCUSSION TITLE
I had to install @floating-ui/react-dom and @floating-ui/dom to get all the types to work

### DISCUSSION THREAD

**Discussion by dagda1** [floating-ui/floating-ui#2536] [Q&A]
When I initially installed `@floating-ui/rect`, I had a lot of types (e.g. `flip`, `offset`, `autoUpdate` etc. resolving to `any`, that after closer inspection were declared in `@floating-ui/react-dom` and `@floating-ui/dom`.

I had to add the following `devDependencies`:

```json
    "@floating-ui/dom": "^1.5.1",
    "@floating-ui/react-dom": "^2.0.2",
```

This makes sense as `@floating-ui/react` has in `types.d.ts` imports from packages that were not initially imported, e.g.

```ts
import type {
  UseFloatingOptions as UsePositionOptions,
  UseFloatingReturn as UsePositionFloatingReturn,
  VirtualElement,
} from '@floating-ui/react-dom';
```

Is this a bug or should these types be bundled with `@floating-ui/react`?

**Comment by atomiks** [comment]
The types are re-exported and those deps are installed transitively anyway. There is something wrong with your setup if they resolve to `any` by default.
